packages/api/portfolioDrafts/funding/createFundingStep.test.ts

```diff
@@ -0,0 +1,418 @@
+const tomorrow = new Date(Date.now() + 24 * 60 * 60 * 1000).toISOString().slice(0, 10);
+
+const ddbMock = mockClient(DynamoDBDocumentClient);
+beforeEach(() => {
+  ddbMock.reset();
+});
+
+const validRequest: APIGatewayProxyEvent = {
+  body: JSON.stringify(mockFundingStepGoodData()),
+  pathParameters: { portfolioDraftId: uuidv4() },
+} as any;
+
+const validRequestBadData: APIGatewayProxyEvent = {
+  body: JSON.stringify(mockFundingStepBadData()),
+  pathParameters: { portfolioDraftId: uuidv4() },
+} as any;
+
+it("sanity check; relative dates used for tests should make sense", () => {
+  expect(new Date(yesterday) < new Date(today)).toEqual(true);
+  expect(new Date(today) < new Date(tomorrow)).toEqual(true);
+});
+
+describe("Handle service level error", function () {
+  it("should return generic Error if exception caught", async () => {
+    jest.spyOn(console, "error").mockImplementation(() => jest.fn()); // suppress output
+    ddbMock.on(UpdateCommand).rejects("Some error occurred");
+    const result = await handler(validRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(DATABASE_ERROR);
+    expect(result.statusCode).toEqual(ErrorStatusCode.INTERNAL_SERVER_ERROR);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+  });
+});
+
+describe("Path parameter tests", function () {
+  it("should require path param", async () => {
+    const emptyRequest: APIGatewayProxyEvent = {} as any;
+    const result = await handler(emptyRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(PATH_PARAMETER_REQUIRED_BUT_MISSING);
+    expect(result.statusCode).toEqual(ErrorStatusCode.BAD_REQUEST);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+  });
+  it("should return error when path param not UUIDv4 (to avoid attempting update)", async () => {
+    const invalidRequest: APIGatewayProxyEvent = {
+      pathParameters: { portfolioDraftId: "invalid" },
+    } as any;
+    const result = await handler(invalidRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(NO_SUCH_PORTFOLIO_DRAFT);
+    expect(result.statusCode).toEqual(ErrorStatusCode.NOT_FOUND);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
+    expect(JSON.parse(result.body).message).toMatch(/Portfolio Draft with the given ID does not exist/);
+  });
+});
+
+describe("Request body tests", function () {
+  it("should return error when request body is empty", async () => {
+    const emptyRequest: APIGatewayProxyEvent = {
+      body: "",
+      pathParameters: { portfolioDraftId: uuidv4() },
+    } as any;
+    const result = await handler(emptyRequest);
+    expect(result).toBeInstanceOf(ErrorResponse);
+    expect(result).toEqual(REQUEST_BODY_EMPTY);
+    expect(result.statusCode).toEqual(ErrorStatusCode.BAD_REQUEST);
+    expect(JSON.parse(result.body).code).toEqual(ErrorCodes.OTHER);
```

Comment:
This is a redundant expectation right? Because if 

```
expect(result).toEqual(REQUEST_BODY_EMPTY);
```

passes, then we know what the HTTP Status code and error code will be right?

Reply:
This is redundant and perhaps overly prescriptive, but I think it's good to include these expectations because the API spec explicitly states them for each combination of operation and response.  It's how I noticed we had `ErrorCodes.INVALID_INPUT` just about everywhere instead of `ErrorCodes.OTHER` and came to understand their intent.  I think there are still instances where INVALID_INPUT is returned incorrectly because the respective unit tests only check the response type.

Also, in some cases the API spec's stated response doesn't squarely fit one in `errors.ts` due to a less common combination of `statusCode` and `code` and `message` (aka response description).

Reply:
For example, incorrect use of INVALID_INPUT can be seen [here](https://github.com/dod-ccpo/atat-web-api/blob/a261d6db9767bb8ed19959edb93e78c9dc525a85/packages/api/portfolioDrafts/portfolio/createPortfolioStep.ts#L13) and [here](https://github.com/dod-ccpo/atat-web-api/blob/a261d6db9767bb8ed19959edb93e78c9dc525a85/packages/api/portfolioDrafts/getPortfolioDrafts.ts#L10) and [here](https://github.com/dod-ccpo/atat-web-api/blob/a261d6db9767bb8ed19959edb93e78c9dc525a85/packages/api/portfolioDrafts/deletePortfolioDraft.ts#L19).

If the respective unit tests were checking the expected `statusCode`, and within the body, `code` and `message` per the API spec that would have eliminated that incorrect use.

Granted all of these examples are where an `ErrorResponse` is instantiated within the function rather than using one in `errors.ts`.  For any `ErrorResponse` from `errors.ts` that we've vetted and proven correct these expectation checks probably are redundant and unnecessary, but I still think they are useful while we're are building and shaking things out.

Reply:
OK that's fair. And I believe it may not technically be redundant until Kyle's fix to tighten up the usages (so that all re-used error types do have the same values for the members that we'd expect like `code`). So I'm good to keep these as-is but eventually would think we ought to remove them so as not to confuse ourselves.

Reply:
#179 has been merged so it should not be possible to have an invalid code here unless you really go out of your way to do so in (which will hopefully be caught during a manual peer review)

Reply:
@kylelaker-ccpo Thanks for addressing INVALID_INPUT/OTHER issue more holistically in #179.